includes/commerce_pos.transaction.inc

```diff
@@ -621,6 +651,32 @@ function commerce_pos_transaction_ajax_check(&$form, &$form_state) {
           $transaction->doAction('save');
           break;
 
+        case 'do-order-search':
+          $order = commerce_order_load($form_state['input']['order_search']['input']);
+          // If we have got a valid order display the item in the order in a
+          // modal.
+          if ($order) {
+            commerce_pos_return_select_item_popup($form, $form_state, $order);
+          }
+          else {
+            form_set_error('', t('The order does not exist.'));
+          }
+
+          break;
+
+        case 'return-items-add':
+          // Go through the return item checkboxes and get the items from the
+          // order that were selected to be returned and add them to the
+          // return transaction.
+          foreach ($form_state['values']['return_items'] as $key => $value) {
+            if (!empty($value)) {
+              // We're sending the actual price the customer paid to the
+              // addProduct() function, so that price will be used in the
+              // transaction instead of the product price.
+              $transaction->addProduct($form_state['triggering_element']['#line_items'][$value]->commerce_product->value(), 1, TRUE, $form_state['triggering_element']['#line_items'][$value]->commerce_unit_price->value()['amount']);
```

Comment:
Instead of sending an integer (amount is either an integer or a floating point) ... we should be sending, at the very least, a pricing array that includes everything from the original price so that currency and [price components](https://drupalcommerce.org/user-guide/price-components). 

But I'll let @smmccabe decide if that is overly complicated for what we're trying to accomplish.